cmd/geth/forgecmd.go

```diff
@@ -1,28 +1,129 @@
 package main
 
 import (
+	"bytes"
 	"context"
+	"encoding/hex"
 	"fmt"
 	"math/big"
+	"os"
+	"reflect"
 	"strings"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/ethereum/go-ethereum/internal/ethapi"
 	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/ethereum/go-ethereum/suave/artifacts"
+	suave_backends "github.com/ethereum/go-ethereum/suave/backends"
+	suave "github.com/ethereum/go-ethereum/suave/core"
+	"github.com/flashbots/go-boost-utils/bls"
+	"github.com/naoina/toml"
 	"github.com/urfave/cli/v2"
 )
 
 var defaultRemoteSuaveHost = "http://localhost:8545"
 
+var (
+	isLocalForgeFlag = &cli.BoolFlag{
+		Name:  "local",
+		Usage: `Whether to run the query command locally`,
+	}
+	whiteListForgeFlag = &cli.StringSliceFlag{
+		Name:  "whitelist",
+		Usage: `The whitelist external endpoints to call`,
+	}
+	ethBackendForgeFlag = &cli.StringFlag{
+		Name:  "eth-backend",
+		Usage: `The endpoint of the confidential eth backend`,
+	}
+	tomlConfigForgeFlag = &cli.StringFlag{
+		Name:  "config",
+		Usage: `The path to the forge toml config file`,
+	}
+)
+
+type suaveForgeConfig struct {
+	Whitelist  []string `toml:"whitelist"`
+	EthBackend string   `toml:"eth_backend"`
+}
+
+func readContext(ctx *cli.Context) (*vm.SuaveContext, error) {
+	// try to read the config from the toml config file
+	cfg := &suaveForgeConfig{}
+
+	if ctx.IsSet(tomlConfigForgeFlag.Name) {
+		// read the toml config file
+		data, err := os.ReadFile(ctx.String(tomlConfigForgeFlag.Name))
+		if err != nil {
+			return nil, err
+		}
+
+		// this is decoding
+		// [profile.suave]
+		var config struct {
+			Profile struct {
+				Suave *suaveForgeConfig
+			}
+		}
+
+		tomlConfig := toml.DefaultConfig
+		tomlConfig.MissingField = func(rt reflect.Type, field string) error {
+			return nil
+		}
+		if err := tomlConfig.NewDecoder(bytes.NewReader(data)).Decode(&config); err != nil {
+			return nil, err
+		}
+		cfg = config.Profile.Suave
+	}
+
+	// override the config if the flags are set
+	if ctx.IsSet(ethBackendForgeFlag.Name) {
+		cfg.EthBackend = ctx.String(ethBackendForgeFlag.Name)
+	}
+	if ctx.IsSet(whiteListForgeFlag.Name) {
+		cfg.Whitelist = ctx.StringSlice(whiteListForgeFlag.Name)
+	}
+
+	// create the suave context
+	var suaveEthBackend suave.ConfidentialEthBackend
+	if cfg.EthBackend != "" {
+		suaveEthBackend = suave_backends.NewRemoteEthBackend(cfg.EthBackend)
+	} else {
+		suaveEthBackend = &suave_backends.EthMock{}
+	}
+
+	ecdsaKey, _ := crypto.GenerateKey()
```

Comment:
Can we check these errors?  It's fine to panic if we "know" they'll never fail.  I've just been bitten too many times by unchecked errors and have become a little neurotic as a result.